wasi/wasi_test.go

```diff
@@ -200,8 +200,210 @@ func TestAPI_ArgsSizesGet_Errors(t *testing.T) {
+		require.Equal(t, ErrnoSuccess, errno)
+		require.Equal(t, expectedMemory, store.Memories[0].Buffer[0:maskLength])
+	})
+	t.Run(FunctionEnvironGet, func(t *testing.T) {
+		maskMemory(store, maskLength)
+
+		ret, _, err := store.CallFunction(context.Background(), "test", FunctionEnvironGet, uint64(resultEnviron), uint64(resultEnvironBuf))
+		require.NoError(t, err)
+		require.Equal(t, ErrnoSuccess, Errno(ret[0])) // cast because results are always uint64
+		require.Equal(t, expectedMemory, store.Memories[0].Buffer[0:maskLength])
+	})
+}
+
+func TestAPI_EnvironGet_Errors(t *testing.T) {
+	envOpt, err := Environ("a=bc", "b=cd")
+	require.NoError(t, err)
+	store, api := instantiateWasmStore(t, FunctionEnvironGet, ImportEnvironGet, "test", envOpt)
+
+	memorySize := uint32(len(store.Memories[0].Buffer))
+	validAddress := uint32(0) // arbitrary valid address as arguments to environ_get. We chose 0 here.
+
+	tests := []struct {
+		name       string
+		environ    uint32
+		environBuf uint32
+	}{
+		{
+			name:       "out-of-memory environPtr",
+			environ:    memorySize,
+			environBuf: validAddress,
+		},
+		{
+			name:       "out-of-memory environBufPtr",
+			environ:    validAddress,
+			environBuf: memorySize,
+		},
+		{
+			name: "environPtr exceeds the maximum valid address by 1",
+			// 4*len(environ.nullTerminatedValues) is the expected buffer size for environPtr, 4 is the size of uint32
+			environ:    memorySize - 4*uint32(len(api.(*wasiAPI).environ.nullTerminatedValues)) + 1,
+			environBuf: validAddress,
+		},
+		{
+			name:       "environBufPtr exceeds the maximum valid address by 1",
+			environ:    validAddress,
+			environBuf: memorySize - api.(*wasiAPI).environ.totalBufSize + 1,
+		},
+	}
+
+	for _, tt := range tests {
+		tc := tt
+
+		t.Run(tc.name, func(t *testing.T) {
+			ret, _, err := store.CallFunction(context.Background(), "test", FunctionEnvironGet, uint64(tc.environ), uint64(tc.environBuf))
+			require.NoError(t, err)
+			require.Equal(t, uint64(ErrnoFault), ret[0]) // ret[0] is returned errno
+		})
+	}
+}
+
+func TestAPI_EnvironSizesGet(t *testing.T) {
+	envOpt, err := Environ("a=b", "b=cd")
+	require.NoError(t, err)
+	resultEnvironc := uint32(1)       // arbitrary offset
+	resultEnvironBufSize := uint32(6) // arbitrary offset
+	maskLength := 11                  // number of bytes to write '?' to tell what we've written
+	expectedMemory := []byte{
+		'?',                // resultEnvironc is after this
+		0x2, 0x0, 0x0, 0x0, // little endian-encoded environment variable count
+		'?',                // resultEnvironBufSize is after this
+		0x9, 0x0, 0x0, 0x0, // little endian-encoded size of null terminated strings
+		'?', // stopped after encoding
+	}
+	store, wasiAPI := instantiateWasmStore(t, FunctionEnvironSizesGet, ImportEnvironSizesGet, "test", envOpt)
+
+	t.Run("API.EnvironSizesGet", func(t *testing.T) {
+		maskMemory(store, maskLength)
+
+		// provide a host context we call directly
```

Comment:
super nit but usually independent comment lines should start with capital letters and end with period here and everywhere 

```suggestion
		// Provide a host context we call directly.
```

Reply:
I'll start to do that also, I think I haven't